### PR TITLE
[Func Test] Upgrade func test of QuantizedConvolutionBatchNorm related to 2.0

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - for easy code review, split from this PR https://github.com/openvinotoolkit/openvino/pull/20901
 - Upgrade func test of QuantizedConvolutionBatchNorm related to 2.0

### Tickets:
 - 124431

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Review by riverlijunjie — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:212]: Mark as `changes requested` because these changes requires additional review before merge

**Review by vurusovs — CHANGES_REQUESTED** [review]

File: src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp
Comment (ilya-lavrenov) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:203]: better to move implementations / template specializations to .cpp files

Comment (zhaixuejun1993) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp]: > better to move implementations / template specializations to .cpp files

@ilya-lavrenov  Yes, found the implementations in cpp by ilya C. Refacted.

File: src/tests/functional/shared_test_classes/include/shared_test_classes/subgraph/quantized_convolution_batch_norm.hpp
Comment (iefode) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/subgraph/quantized_convolution_batch_norm.hpp:32]: minor: align

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: Could you please check all fake quantize values? I am not sure that all used values are correct...

**Review by iefode — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]
Please, address @iefode comments

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp
Comment (zhaixuejun1993) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: @iefode Here change the range value to 10, according to the old workflow, which using the default value at here:
https://github.com/openvinotoolkit/openvino/blob/61c5a80fa63ed3dd7eb353c62ceee908ef9d8d69/src/tests/test_utils/functional_test_utils/include/functional_test_utils/blob_utils.hpp#L504-L508
Another reason is without this change, the following test about "*QuantizedConvolutionBatchNorm*" will failed, such as
image
I don't know how to check all fake quantize values. Can u share more details?

Comment (iefode) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: Which details do you mean? It is parameters to generate input tensors for fake quantize.. I don't know fake quantize details implementation. It is the main reason why I have asked you

Comment (zhaixuejun1993) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: Yes, this parameter is for fake quantize input generate. I compared the API 1.0 input data & API 2.0 input data, with this change can get the same input. And all tests related quantize can pass. So I think we need this change to keep the same as API 1.0. Here is a question, why for quantize operation the inputs need to be limited in 0~10? I'm not very familiar with quantize operation, anybody can help clarify this?

Comment (zhaixuejun1993) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: @vurusovs can we merge this PR first? And create another ticket to track why for "BatchNormInference" operation the input range need to be limited to 0~10.

Comment (vurusovs) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: Because inputs generation is shared between different types of tests, we need confirmation there is no impact  on other tests in terms of coverage
Waiting for @iefode and @eshoguli for confirmation

Comment (iefode) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: @zhaixuejun1993 
I will discuss the ranges with @eshoguli as a  `fakeQuantize` expert.
Will provide the technical update in the PR comment.
Please do not merge before update

Comment (zhaixuejun1993) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: > @zhaixuejun1993 I will discuss the ranges with @eshoguli as a `fakeQuantize` expert. Will provide the technical update in the PR comment. Please do not merge before update

That will be great! I will also do some learning about fakeQuantize

Comment (eshoguli) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: @zhaixuejun1993, Ideally we should define data interval for `FakeQuantize` based on input intervals (2nd and 3rd inputs). If we have few failed tests then we can use static variables to use previous intervals to fix tests. If we have a lot of failed tests, then we can update intervals later. Thanks!

Comment (zhaixuejun1993) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: @eshoguli Only the tests about "QuantizedConvolutionBatchNorm" failed. Maybe we can use previous intervals to fix this tests.

Comment (iefode) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:223]: @zhaixuejun1993 
We had a discussion with @eshoguli as LPT transformation owner. As he said, we will ask you to fix the failed tests in case it is a couple of tests to merge PR. In other case: our team will update the intervals for `fakeQuantize` as a part of LPT OV2.0 migration process. Thanks!

**Pull request merged**
